autosklearn/metalearning/metafeatures/metafeatures.py

```diff
@@ -267,24 +267,32 @@ def _calculate(self, X, y, logger, categorical):
 # TODO: generalize this!
 @metafeatures.define("NumberOfNumericFeatures")
 class NumberOfNumericFeatures(MetaFeature):
-    def _calculate(self, X, y, logger, categorical):
-        return len(categorical) - np.sum(list(categorical.values()))
+    def _calculate(self, X, y, logger, feat_type):
+        categorical = {
```

Comment:
Could you please make this future-proof by directly counting the number of numerical features (as for the categorical features below)?